stubs/SQLAlchemy/sqlalchemy/util/_collections.pyi

```diff
@@ -16,9 +16,9 @@ collections_abc = collections.abc
 EMPTY_SET: frozenset[Any]
 
 class ImmutableContainer:
-    def __delitem__(self, *arg: object, **kw: object) -> NoReturn: ...
-    def __setitem__(self, *arg: object, **kw: object) -> NoReturn: ...
-    def __setattr__(self, *arg: object, **kw: object) -> NoReturn: ...
+    def __delitem__(self, *arg: Unused, **kw: Unused) -> NoReturn: ...
+    def __setitem__(self, *arg: Unused, **kw: Unused) -> NoReturn: ...
+    def __setattr__(self, *arg: Unused, **kw: Unused) -> NoReturn: ...
```

Comment:
(I haven't looked at the source code here, just speculating.)

If instances of `ImmutableContainer` are meant to be, well, immutable, it might produce better type checking if we just omitted these methods from the stub altogether.

Reply:
Maybe relevant to your question:
https://github.com/microsoft/pyright/issues/4653
https://github.com/python/mypy/issues/14726